src/coreclr/vm/codeman.cpp

```diff
@@ -1414,6 +1414,17 @@ void EEJitManager::SetCpuInfo()
             {
                 CPUCompileFlags.Set(InstructionSet_AVX10v1_V512);
             }
+            if ((cpuFeatures & XArchIntrinsicConstants_Avx10v2) != 0)
+            {
+                if (CLRConfig::GetConfigValue(CLRConfig::EXTERNAL_EnableAVX10v2))
+                {
+                    CPUCompileFlags.Set(InstructionSet_AVX10v2);
+                    if((cpuFeatures & XArchIntrinsicConstants_Avx512) != 0)
+                    {
+                        CPUCompileFlags.Set(InstructionSet_AVX10v2_V512);
+                    }
+                }
+            }
```

Comment:
nit: We normally don't nest the top level ISA checks to reduce the overall nesting and improve readability.

Instead, we rely on the `CPUCompileFlags.EnsureValidInstructionSetSupport` call down below to ensure they're consistent if some environment variable disabled a dependency.

Reply:
I have updated the PR accordingly.